postgis/lwgeom_export.c

```diff
@@ -393,13 +244,7 @@ Datum LWGEOM_asGeoJson(PG_FUNCTION_ARGS)
 	}
 
 	lwgeom = lwgeom_from_gserialized(geom);
-	geojson = lwgeom_to_geojson(lwgeom, srs, precision, output_bbox);
-	lwgeom_free(lwgeom);
-
-	if (srs) pfree(srs);
-
-	PG_FREE_IF_COPY(geom, 0);
-	PG_RETURN_TEXT_P(geojson);
+	PG_RETURN_TEXT_P(lwgeom_to_geojson(lwgeom, srs, precision, output_bbox));
```

Comment:
lwgeom_free(lwgeom); ?

Reply:
Probably okay since we're just before leaving a function and context is going to die anyway, but I nervously rechecked it's in postgis/ and not lwgeom/.

Reply:
Anything except index support functions you get the free "deallocation on exit".

Reply:
Yes, I've been doing this lately in multiple postgis functions (the ones were liblwgeom now returns a lwvarlena_t object). I think it keeps things cleaner and it's good for performance as it's faster to wait for the deallocation of the context.